pyccl/tracers.py

```diff
@@ -264,6 +296,133 @@ def get_bessel_derivative(self):
+                # Scale-dependent case with full k array
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1],
+                                   (bias_transfer_k[1] * mg_transfer[2]))
+                self.add_tracer(cosmo, kernel, transfer_ka=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles)
+            elif (isinstance(k, float) or isinstance(k, int)):
+                # Scale-dependent case with  a single specified k value
+                self.add_tracer(cosmo, kernel, transfer_a=mg_transfer,
+                                der_bessel=der_bessel, der_angles=der_angles,
+                                transfer_k=bias_transfer_k)
+
+    def _get_MG_transfer_function(self, cosmo, z):
+        """ This function allows to obtain the function Sigma(z,k) (1 or 2D
+            arrays) for an array of redshifts coming from a redshift
+            distribution (defined by the user) and a single value or
+            an array of k specified by the user. We obtain then Sigma(z,k) as a
+            1D array for those z and k arrays and then convert it to a 2D array
+            taking into consideration the given sizes of the arrays for z and k
+            The MG parameter array goes then as a multiplicative factor within
+            the MG transfer function. If k is not specified then only a 1D
+            array for Sigma(a,k=0) is used.
+
+        Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): cosmology object used to
+                transform redshifts into distances.
+            z (float or tuple of arrays): a single z value (e.g. for CMB)
+                or a tuple of arrays (z, N(z)) giving the redshift distribution
+                of the objects. The units are arbitrary; N(z) will be
+                normalized to unity.
+            k (float or array): a single k value or an array of k for which we
+                calculate the MG parameter Sigma(a,k). For now, the k range
+                should be limited to linear scales.
+        """
+        # Sampling scale factor from a very small (at CMB for example)
+        # all the way to 1 here and today for the transfer function.
+        # For a < a_single it is GR (no early MG)
+        if isinstance(z, float):
+            a_single = 1/(1+z)
+            a = np.linspace(a_single, 1, 100)
+            # a_single is for example like for the CMB surface
+        else:
+            if z[0] != 0.0:
+                stepsize = z[1]-z[0]
+                samplesize = int(z[0]/stepsize)
+                z_0_to_zmin = np.linspace(0.0, z[0] - stepsize, samplesize)
+                z = np.concatenate((z_0_to_zmin, z))
+            a = 1./(1.+z)
+        a.sort()
+        # Scale-dependant MG case with an array of k
+        nk = lib.get_pk_spline_nk(cosmo.cosmo)
+        status = 0
+        k, status = lib.get_pk_spline_lk(cosmo.cosmo, nk, status)
+        check(status)
+        k = np.exp(k)
```

Comment:
If you change the previous lines to:
```
        lk, status = lib.get_pk_spline_lk(cosmo.cosmo, nk, status)
        check(status)
        k = np.exp(lk)
```
you'll have log(k) already computed (so you save on calculating one extra spurious logarithm). See my next comment.

Reply:
Agreed. Done!